### ISSUE TITLE
Function pam.read_file is not available?

### ISSUE THREAD

**Issue by lorengordon** [saltstack/salt#20809]
I've been helping work on a security baseline implemented via salt and saw that salt has the start of a pam module. But it appears salt will only load the module if `libpam.so` exists at `/usr/lib/libpam.so`. On a CentOS system with the 64bit version of the pam rpm, the corresponding file exists in `/lib64`. I'm not sure that `os.path.exists` will have the desired behavior anyway, though, since the file is really `/lib64/libpam.so.0` and is a symlink to `/lib64/libpam.so.0.82.2`. Any idea why this loading restriction is in there?

https://github.com/saltstack/salt/blob/develop/salt/modules/pam.py
```
def __virtual__():
    '''
    Only load the module if iptables is installed
    '''
    if os.path.exists('/usr/lib/libpam.so'):
        return 'pam'
    return False
```

```
# salt-call --local pam.read_file /etc/pam.d/system-auth
Function pam.read_file is not available
```

**Comment by rallytime** [comment]
Thanks for the report @lorengordon. I am not very familiar with the particulars of the pam module, but it looks like @techhat can perhaps comment on your question here.

**Comment by rallytime** [comment]
@lorengordon Apologies for never circling back around to your question. I think this restriction was just one way to check if pam is installed when the module was first written. If there are other file paths that would expose the installation of pam, that virtual check can certainly be adjusted.

**Comment by lorengordon** [comment]
Heh, no worries. This looks like a pretty easy one. I'll try to come back to it and get it working.

**Comment by rallytime** [comment]
That would be great. I'll change this to a more "this should be addressed" labeling scheme rather than just "Question". :)

**Comment by rallytime** [comment]
Fixed via #34002
